CordovaLib/Classes/Public/CDVViewController.m

```diff
@@ -720,12 +720,25 @@ - (void)onAppWillTerminate:(NSNotification*)notification
     }
 }
 
+- (void)checkAndReinitViewUrl
+{
+    if ([[[self.webViewEngine URL] absoluteString] isEqualToString:@"about:blank"]) {
```

Comment:
In the unlikely scenario that `appURL` is actually `about:blank`, you should check for this condition.
RIght now there is no probability of an infinite loop, but you never know if there are side-effects with any calls further down the stack in `checkAndReinitViewUrl `

I would return a `BOOL` from this function (`YES` if it re-inited, 'NO' if it did not), and add a unit test in https://github.com/apache/cordova-ios/blob/master/tests/CordovaLibTests/CDVViewControllerTest.m

Reply:
thanks! I'll try to update these when I have a chance :) we've found another case when testing that the loaded url in the webview is (null), so added those checks too